### ISSUE TITLE
no indication if tox-docker plugin is installed or used

### ISSUE THREAD

**Issue by ssbarnea** [dcrosta/tox-docker#1]
First please allow me congrats you for writing this, I was considering doing this myself but at some point but lack of time didn't even allow me to attempt. 

Maybe now I could be of help as I find this extremely useful. I am an OpenStack engineer but my main working machine is a MacOS one which means that even if most OpenStack components are Python based I will encounter lots of problems running tests. By being able to run tests in docker I will be able to edit the code on mac but to run the tests in a supported container.

After I did `pip install tox-docker` and added the two config lines in `tox.ini` I expected to be able to use the it. 

Still, nothing happened, `tox -V"

```
tox --version                                                                                                                                          2.7.0 imported from /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/tox/__init__.pyc

tox --help
# help output with additional options added by optional tox-pyenv, but no indication that tox-docker was loaded

pip freeze|grep tox                                                                                                                                    detox==0.10.0
tox==2.7.0
tox-docker==1.0.0
tox-pyenv==1.0.3
```
The only place where I was able to find that `tox-docker` was installed was while running `tox --hi` which displayed the two additional ini options.

Still, it seems a bit weird not to be able to enable or disable the docker use with a command line option or environment variable.

**Comment by dcrosta** [comment]
I agree it would be nice if `tox --help` would show you which tox plugins were detected, but that's out of control of the plugins themselves. Sounds like a good feature request against https://github.com/tox-dev/tox though.

As far as tox-docker is concerned, what command line flags would you want or expect? There's currently no optional behavior in tox docker. I suppose you could have a flag to disable it, but that would just mean your tests would fail, since whatever they are looking for in a docker container wouldn't be there, right?

**Comment by ssbarnea** [comment]
I raised https://github.com/tox-dev/tox/issues/544 regarding displaying `tox --version` not displaying plugins.

Now regarding flags, I would see something like `--docker auto|true|false` with default behaviour of auto-detecting it. At this moment it seems that tox-docker is not using docker if docker is not started. I had invalid settings for ``DOCKER_HOST`` and ``DOCKER_API_VERSION`` on my machine but tox did run without any complaints.

After fixing vars it did work. Try with something like `DOCKER_HOST=tcp://unresolving.domain:1234` to replicate. If you put 127.0.0.1 and the port is not open you do get a more or less nice exception with the clear message. I didn't get that on the first case. 

Anyway, I would mark this bug as **trivial**.

**Comment by dcrosta** [comment]
Ah, so you're saying some people might want to run tests with tox sometimes using docker and sometimes not (ie. using locally-running, "real" versions of whatever would have been dockerized) -- OK, that makes sense as a feature request, and is not something I had considered. Right now the best way to accomplish that would be by uninstalling tox-docker when you want to not use docker, and reinstall it when you do. Obviously that's pretty annoying, so if you want to scratch that itch and submit a change set, I'll be happy to look at that :)

**Comment by ssbarnea** [comment]
This was fixed in tox 2.9.0, which now reports all installed plugins with `tox --version`.

**Comment by dcrosta** [comment]
Cool -- thanks for looping back here!
